roles/rsyslog/tasks/main.yml

```diff
@@ -54,6 +54,22 @@
         - rsyslog_unprivileged | bool
         - rsyslog_user != 'root'
 
+    # Checking if the config directory exists or not
+    - name: Get config directory stat
+      stat:
+        path: "{{ rsyslog_config_dir }}"
+      register: stat_rsyslog_config_dir
```

Comment:
This part should not be necessary - the `file` module should only create the directory if it does not exist: https://docs.ansible.com/ansible/latest/modules/file_module.html#file-module
```yaml
- name: Create a directory if it does not exist
  file:
    path: /etc/some_directory
    state: directory
    mode: '0755'
```

Reply:
and, fyi - even though you are getting rid of this - the variable you created with `register` should be named appropriately according to the best practices - https://github.com/oasis-roles/meta_standards/pull/12/files#diff-04c6e90faac2675aa89e2176d2eec7d8R81 (soon to be merged) - so something like `register: __rsyslog_register_config_dir`

Reply:
> This part should not be necessary - the `file` module should only create the directory if it does not exist: https://docs.ansible.com/ansible/latest/modules/file_module.html#file-module
> 
> ```yaml
> - name: Create a directory if it does not exist
>   file:
>     path: /etc/some_directory
>     state: directory
>     mode: '0755'
> ```

Thank you, @richm.  I removed the unnecessary stat task.